eventually/authentication/views.py

```diff
@@ -2,13 +2,76 @@
 Views module
 ============
 """
-
+import json
+from json.decoder import JSONDecodeError
+from django.http import HttpResponse
 from django.views.generic.base import View
+from eventually.settings import FRONT_HOST
+from utils.jwttoken import create_token, handle_token
+from utils.send_mail import send_email
+from authentication.models import CustomUser
+from utils.validators import registration_user_validate
 
 
 class UserView(View):
     """UserView view handles GET, POST, PUT, DELETE requests."""
 
 
-def registration_user():
+def registration_user(request):
     """Registration CustomUser"""
+
+    if request.method == 'POST':
+        try:
+            data = json.loads(request.body.decode('utf-8'))
+        except JSONDecodeError:
+            return HttpResponse(status=400)
+
+        data = {
+            'first_name': data.get('first_name'),
+            'last_name': data.get('last_name'),
+            'middle_name': data.get('middle_name'),
+            'email': data.get('email'),
+            'password': data.get('password')
+        }
+        if not registration_user_validate(data):
+            return HttpResponse('Data is not valid', status=400)
+
+        user = CustomUser.get_by_email(data['email'])
+        if user:
+            return HttpResponse('Email is already exist', status=400)
+
+        user = CustomUser.create(first_name=data['first_name'],
+                                 last_name=data['last_name'],
+                                 middle_name=data['middle_name'],
+                                 email=data['email'],
+                                 password=data['password'])
+        if not user:
+            return HttpResponse(status=400)
+
+        ctx = {
+            'first_name': user.first_name,
+            'domain': FRONT_HOST,
+            'token': create_token({'email': user.email}),
+        }
+        message = 'registration'
+        mail_subject = 'Activate account'
+        send_email(mail_subject, message, [user.email], 'registration.html', ctx)
+
+        return HttpResponse('Please confirm your email address to complete the registration', status=201)
+
+
+def activate(request, token):
+    """Activation CustomUser"""
+    if request.method == 'GET':
+        try:
+            email = handle_token(token)
+            user = CustomUser.objects.get(email=email['email'])
+        except(TypeError, ValueError, OverflowError, CustomUser.DoesNotExist):
+            user = None
+        if user is not None:
+            user.is_active = True
+            user.save()
+            success_msg = 'Thank you for email confirmation. Now you can login your account.'
+            return HttpResponse(success_msg, status=200)
+
+        return HttpResponse('Activation link is invalid!')
```

Comment:
If the request method is other than GET, shouldn't we return something in that case?